File: app/src/main/java/com/pleon/buyt/viewmodel/CreateStoreViewModel.java
package com.pleon.buyt.viewmodel;

import android.app.Application;

import com.pleon.buyt.database.repository.StoreRepository;
import com.pleon.buyt.model.Store;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

public class CreateStoreViewModel extends AndroidViewModel {

    private StoreRepository mStoreRepository;

    public CreateStoreViewModel(Application application) {
        super(application);
        mStoreRepository = StoreRepository.getInstance(application);
    }

    public LiveData<Store> addStore(Store store) {
        return mStoreRepository.insert(store);
    }
}

File: app/src/main/kotlin/com/pleon/buyt/viewmodel/CreateStoreViewModel.kt
package com.pleon.buyt.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.pleon.buyt.model.Store
import com.pleon.buyt.repository.StoreRepository

class CreateStoreViewModel(app: Application, private val repository: StoreRepository)
    : AndroidViewModel(app) {

    suspend fun addStore(store: Store) = repository.insert(store)
}
